streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -285,18 +306,13 @@ void handleLostAll() {
         final Iterator<Task> iterator = tasks.values().iterator();
         while (iterator.hasNext()) {
             final Task task = iterator.next();
-            final Set<TopicPartition> inputPartitions = task.inputPartitions();
             // Even though we've apparently dropped out of the group, we can continue safely to maintain our
             // standby tasks while we rejoin.
             if (task.isActive()) {
+                cleanupTask(task);
                 task.closeDirty();
-                changelogReader.remove(task.changelogPartitions());
-            }
-
-            for (final TopicPartition inputPartition : inputPartitions) {
-                partitionToTask.remove(inputPartition);
+                iterator.remove();
```

Comment:
Why do we only remove the task if it was active now?

Reply:
We previously also only remove the task (https://github.com/apache/kafka/pull/8058#discussion_r377123490), the original motivation to only remove the task is that standby tasks are likely to go back and even if they do not they can still be closed in the next rebalance's onAssignment call, so we just keep them a bit longer hoping it worth the "wait" :)